### DISCUSSION TITLE
New device request?

### DISCUSSION THREAD

**Discussion by JonScaife** [custom-components/ble_monitor#767] [Ideas]
Hi
Thanks for the great work on this.  I'm using it to monitor BLE plant sensors all over the house and it works really well.
I wondered if you were accepting requests for other devices to consider supporting?  I have an "Air Mentor Pro 2" which uses BLE.  Someone has got it working with Samsungs "SmartThings" platform and I was able to connect to it from the bluetooth command line within HA, but my distinct lack of programming knowledge seems to have been too big a barrier to actually pulling meaningful data from it.  If you are accepting requests please consider this device!
The code for the SmartThings implementation is at https://community.smartthings.com/t/successfully-integrated-air-quality-monitor-air-mentor-pro-2-in-smartthings-using-raspberry-pi/79574
Many thanks

**Comment by Ernst79** [comment]
Yes, looks very well possible. Based on the info in the link, I think it sends two messages. 

**first message**

`043E1B02010000A5808FE648540F0201060BFF222100b91963332d0145CC`

I managed to translate this to:

```
result  ({'TVOC': 185, 'temperature': 24.99, 'temperature calibrated': 19.89, 'humidity': 61.34, 'IAQ': 325, 'rssi': -52, 'mac': '5448E68F80A5', 'type': 'Air Mentor Pro 2', 'packet': 'no packet id', 'firmware': 'Air Mentor', 'data': True}, None)
```

**second message** 

`043E1B02010000A5808FE648540F0201060BFF21212710000300030000CC`

```
result  ({'co2': 10000, 'pm25': 3, 'pm10': 3, 'battery': 0, 'voltage': 0, 'rssi': -52, 'mac': '5448E68F80A5', 'type': 'Air Mentor Pro 2', 'packet': 'no packet id', 'firmware': 'Air Mentor', 'data': True}, None)
```

The battery and voltage sensor are most likely something else, the python script in the link you gave doesn't use this. My guess one of these is is something like "charging" or "connected to the main power" I will create a test version for you to find out. coming days. .

**Comment by Ernst79** [comment]
Moved this to a new issue, please discuss further in the issue #768
